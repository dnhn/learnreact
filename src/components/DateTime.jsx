import React, { PureComponent } from 'react';

export default class DateTime extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { dateTime: this.newDateTime() };
  }

  componentDidMount = () => this.tick();

  newDateTime = () => `${new Date().toLocaleDateString()} ${new Date().toLocaleTimeString()}`;

  tick = () => setInterval(() => this.setState({ dateTime: this.newDateTime() }), 1000);

  render = () => <div style={{ padding: '1% 0' }}>{this.state.dateTime}</div>;
}
