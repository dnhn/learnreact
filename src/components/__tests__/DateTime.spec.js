import React from 'react';
import { render } from '@testing-library/react';
import DateTime from '../DateTime';

it('renders correctly', () => {
  const { asFragment } = render(<DateTime />);
  expect(asFragment()).toMatchSnapshot();
});
