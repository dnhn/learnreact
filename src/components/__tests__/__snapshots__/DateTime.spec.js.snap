// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    style="padding: 1% 0px;"
  >
    3/29/2020 7:50:23 PM
  </div>
</DocumentFragment>
`;
